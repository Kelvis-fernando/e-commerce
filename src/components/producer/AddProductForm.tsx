// import axios from "axios";
import useAddProductForm from "../../hooks/producer/useAddProductForm";
import {
  AddProductFormContainer,
  InputStyles,
} from "../../styles/producer/addProductForm";

const AddProductForm = () => {
  const { handleCreateProduct } = useAddProductForm();

  return (
    <AddProductFormContainer onSubmit={handleCreateProduct}>
      <h1>Cadastre o produto</h1>
      <InputStyles>
        <div>
          <label htmlFor="name">Nome</label>
          <input type="text" name="name" id="name" />
        </div>
        <div>
          <label htmlFor="brand">Marca</label>
          <input type="text" name="brand" id="brand" />
        </div>
      </InputStyles>
      <InputStyles>
        <div>
          <label htmlFor="type">Tipo</label>
          <select name="type" id="type">
            <option value="Fraco">Fraco</option>
            <option value="Medio">Medio</option>
            <option value="Forte">Forte</option>
          </select>
        </div>
        <div>
          <label htmlFor="price">Preço</label>
          <input type="number" name="price" id="price" />
        </div>
      </InputStyles>
      <InputStyles>
        <div>
          <label htmlFor="classification">Classificação</label>
          <input type="text" name="classification" id="classification" />
        </div>
        <div>
          <label htmlFor="intensity">Intensidade</label>
          <input type="text" name="intensity" id="intensity" />
        </div>
      </InputStyles>
      <InputStyles>
        <div>
          <label htmlFor="origin">Origem</label>
          <input type="text" name="origin" id="origin" />
        </div>
        <div>
          <label htmlFor="typeRoast">Tipo de torra</label>
          <input type="text" name="typeRoast" id="typeRoast" />
        </div>
      </InputStyles>
      <InputStyles>
        <div>
          <label htmlFor="notes">Notas</label>
          <input type="text" name="notes" id="notes" />
        </div>
        <div>
          <label htmlFor="quantity">Quantidade em estoque</label>
          <input type="number" name="quantity" id="quantity" />
        </div>
      </InputStyles>
      <InputStyles>
        <div>
          <label htmlFor="description">Descrição</label>
          <input type="text" name="description" id="description" />
        </div>
        <div>
          <label htmlFor="image">Image</label>
          <input type="text" name="image" id="image" />
        </div>
      </InputStyles>
      <button type="submit">ADICIONAR</button>
    </AddProductFormContainer>
  );
};

export default AddProductForm;
