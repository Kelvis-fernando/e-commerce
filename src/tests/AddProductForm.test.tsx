import React from "react";
import { render } from "@testing-library/react";
import AddProductForm from "../components/producer/AddProductForm";

describe("AddProductForm", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<AddProductForm />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
