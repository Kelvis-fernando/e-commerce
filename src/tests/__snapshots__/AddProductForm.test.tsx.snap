// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProductForm should render and match snapshot 1`] = `
<div>
  <form
    class="sc-bcXHqe gYwOoV"
  >
    <h1>
      Cadastre o produto
    </h1>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="name"
        >
          Nome
        </label>
        <input
          id="name"
          name="name"
          required=""
          type="text"
        />
      </div>
      <div>
        <label
          for="brand"
        >
          Marca
        </label>
        <input
          id="brand"
          name="brand"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="type"
        >
          Tipo
        </label>
        <select
          id="type"
          name="type"
        >
          <option
            value="Fraco"
          >
            Fraco
          </option>
          <option
            value="Medio"
          >
            Medio
          </option>
          <option
            value="Forte"
          >
            Forte
          </option>
        </select>
      </div>
      <div>
        <label
          for="price"
        >
          Preço
        </label>
        <input
          id="price"
          name="price"
          required=""
          type="number"
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="classification"
        >
          Classificação
        </label>
        <input
          id="classification"
          name="classification"
          required=""
          type="text"
        />
      </div>
      <div>
        <label
          for="intensity"
        >
          Intensidade
        </label>
        <input
          id="intensity"
          name="intensity"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="origin"
        >
          Origem
        </label>
        <input
          id="origin"
          name="origin"
          required=""
          type="text"
        />
      </div>
      <div>
        <label
          for="typeRoast"
        >
          Tipo de torra
        </label>
        <input
          id="typeRoast"
          name="typeRoast"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="notes"
        >
          Notas
        </label>
        <input
          id="notes"
          name="notes"
          required=""
          type="text"
        />
      </div>
      <div>
        <label
          for="quantity"
        >
          Quantidade em estoque
        </label>
        <input
          id="quantity"
          name="quantity"
          required=""
          type="number"
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="description"
        >
          Descrição
        </label>
        <input
          id="description"
          name="description"
          required=""
          type="text"
        />
      </div>
      <div>
        <label
          for="image"
        >
          Image
        </label>
        <input
          id="image"
          name="image"
          required=""
          type="text"
        />
      </div>
    </div>
    <button
      type="submit"
    >
      ADICIONAR
    </button>
  </form>
</div>
`;
